proto/xmtpv4/message_api/message_api.proto

```diff
@@ -53,7 +52,6 @@ message UnsignedOriginatorEnvelope {
 // An alternative to a signature for blockchain payloads
 message BlockchainProof {
   uint64 block_number = 1;
-  uint32 publisher_node_id = 2;
```

Comment:
We could also include the transaction hash, since this proof gets generated when the message is read off the chain. That would make it easier to verify

Reply:
Do we expect recipients to verify both the `block_number` and `transaction_hash`? Or would the `transaction_hash` alone be sufficient to find the proof? Just trying to avoid redundant fields that need to be kept consistent where possible

Reply:
(Also, you can merge first, I'll apply this on top of your change)

Reply:
Think the hash could replace the block number, since you can get the block number with the hash